storey/flow.py

```diff
@@ -1649,6 +1667,28 @@ async def _do(self, event):
             results: list[_ParallelExecutionRunnableResult] = await asyncio.gather(*futures)
             if len(self.runnables) == 1:
                 event.body = results[0].data if results else None
+                if self.monitored:
+                    event.monitoring_data = (
+                        {
+                            "microsec": results[0].runtime,
+                            "when": results[0].timestamp.isoformat(sep=" ", timespec="microseconds"),
+                            "error": event.body.get("error") if isinstance(event.body, dict) else None,
+                        },
+                    )
             else:
                 event.body = {result.runnable_name: result.data for result in results}
+                if self.monitored:
+                    monitoring_data = {
+                        result.runnable_name: {
+                            "microsec": result.runtime,
+                            "when": result.timestamp.isoformat(sep=" ", timespec="microseconds"),
+                            "error": (
+                                event.body.get(result.runnable_name).get("error")
+                                if isinstance(event.body.get(result.runnable_name), dict)
+                                else None
+                            ),
+                        }
+                        for result in results
+                    }
+                    event.monitoring_data = monitoring_data
```

Comment:
I guess there's no reason to assign indirectly?
```suggestion
                if self.monitored:
                    event.monitoring_data = {
                        result.runnable_name: {
                            "microsec": result.runtime,
                            "when": result.timestamp.isoformat(sep=" ", timespec="microseconds"),
                            "error": (
                                event.body.get(result.runnable_name).get("error")
                                if isinstance(event.body.get(result.runnable_name), dict)
                                else None
                            ),
                        }
                        for result in results
                    }
```

Reply:
Also wondering if it should be `event._monitoring_data` since we don't want the user rummaging through it.